### PR TITLE
LPD-15313 BuildLang

### PR DESCRIPTION
**Pull request by veroglez**
Hi guys, 

In Echo team we are working on a new functionality in editing Web Contents to identify non-localizable fields more easily (https://liferay.atlassian.net/browse/LPD-6819). Therefore, for each non-localizable field, when the language changes, an info icon has been added next to the label with a tooltip, indicating that the field is not localizable, and also the color of the label changes.

https://github.com/liferay-forms/liferay-portal/assets/9701095/7c98738a-7c33-45bd-8626-cd251dc67b49

Improvements have also been made regarding accessibility:
- A tooltip should be opened with a click or keydown events and for this reason the element that launches the tooltip has been changed to a button (https://github.com/liferay-forms/liferay-portal/commit/ad252bcbe8a9d9244e5848749dc5f4c66e732c1f).
- Change a `fieldset` to a `<div role="group" aria-labelledby={labelId}>` and `legend` to `label`. In terms of semantics it is the same and this technique provides additional styling possibilities (https://github.com/liferay-forms/liferay-portal/commit/638f650dd97c239cb03d8b1e1faef67060b84821).

The Feature Flag to test it is `feature.flag.LPS-114700=true`

Thanks in advance! :smile:

### CODE REVIEW

**Comment by rebsilva** [comment]
just started reviewing! :cherry_blossom:

**Review by rebsilva — COMMENTED** [review]
hi @veroglez! just reviewed and everything seems fine, just some things i want to point out:
- i left a comment about a `const` on the code that maybe will need to be renamed, please let me know what you think!
- i tested this on liferay CE and is not working but in dxp it is, i guess this is expected right?
- i predict that a lot of poshi tests will fail because of the changes on the HTML structure, (the tags and the classes)

i will run the tests so we can already see what tests will fail; if you have any questions please let me know, thanks! :cherry_blossom:

**Comment by rebsilva** [comment]
ci:test:sf

**Comment by rebsilva** [comment]
ci:test:relevant

**Comment by rebsilva** [comment]
ci:test:data-engine

**Comment by marcelabc** [comment]
Hey @veroglez!
Let us know if you have any comment/question about Rebeca's review, okay?
Thank you!

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.js [hunk 1]
```diff
@@ -367,12 +368,13 @@ export function FieldBase({
 			{renderLabel && (
 				<>
 					{showLegend ? (
```
[the remark this answers is not in the document]
Comment (veroglez) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/FieldBase/ReactFieldBase.es.js:382]: Yes, I think is more appropiate change that variable to `showGroup`

**Comment by veroglez** [comment]
Hi 

> hi @veroglez! just reviewed and everything seems fine, just some things i want to point out:
> 
> * i left a comment about a `const` on the code that maybe will need to be renamed, please let me know what you think!
> * i tested this on liferay CE and is not working but in dxp it is, i guess this is expected right?
> * i predict that a lot of poshi tests will fail because of the changes on the HTML structure (the tags and the classes)
> 
> i will run the tests so we can already see what tests will fail; if you have any questions please let me know, thanks! 🌸

Hi @rebsilva, thanks for your review and feeback! Answering to your comments:

- Yes, I think it's more apropiate replace the variable `showLegend` by `showGroup`. I have added this commit 38af7d2e86b99a6226eb1426b01920805127f970
- I just tested in DXP and CE and everything seems to work the same. Is there any issue you can report to see what's happening?
- It seems that there are tests that are failing but I don't know if that tests are related. Can you help me check it?

**Comment by rebsilva** [comment]
hello @veroglez! i will review again and also take a look at the tests failing

**Comment by rebsilva** [comment]
ci:test:forms

**Comment by rebsilva** [comment]
ci:test:data-engine

**Comment by rebsilva** [comment]
hi @veroglez! :cherry_blossom: 

thanks for changing the name for `showGroup`, i think it looks great like this; i also tested the PR again on CE and DXP and both are working, maybe it was something on my computer that wasn't right before.

about the tests i analyzed that many have related failures, because the classes/tags changed are now different from what the poshi locators are looking for. they are:

_[forms suite](https://testray.liferay.com/home/-/testray/case_results?testrayBuildId=2536555491&testrayRunId=2536563560)_: about 28 tests failing

`Element is not present at "//legend......`
`contains(@class,'ddm-field')]//legend[contains(@class,'lfr-ddm-legend')]........`

_[data-engine suite](https://testray.liferay.com/home/-/testray/case_results?testrayBuildId=2536609472&testrayRunId=2536609529)_: about 103 tests failing

same errors as forms suite.

_[relevant suite](https://testray.liferay.com/home/-/testray/case_results?testrayBuildId=2534119879&testrayRunId=2534119938)_: about 18 tests failing

same errors as forms suite.

i think you just need to change two or three paths on poshi and the tests will be fixed, but let me know if you need help with this!

**Review by rebsilva — CHANGES_REQUESTED** [review]
everything looks great on the code, but needs the fix for the related tests failing

**Comment by veroglez** [comment]
ci:test:forms

**Comment by veroglez** [comment]
ci:test:data-engine

**Comment by rebsilva** [comment]
ci:test:relevant

**Comment by rebsilva** [comment]
ci:test:sf

**Comment by rebsilva** [comment]
ci:test:data-engine

**Comment by rebsilva** [comment]
ci:test:forms

**Comment by rebsilva** [comment]
some tests on the relevant suite are still failing and seem related:

[FormsTranslations#GridFieldCanBeTranslated](https://testray.liferay.com/home/-/testray/case_results/2537164155?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2537163646%26testrayRunId%3D2537163696%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)
[FormsGridField#OptionsCanBeDeleted](https://testray.liferay.com/home/-/testray/case_results/2537164056?redirect=https%3A%2F%2Ftestray.liferay.com%2Fhome%2F-%2Ftestray%2Fcase_results%2Findex%3FtestrayBuildId%3D2537163646%26testrayRunId%3D2537163696%26cur%3D1%26delta%3D200%26orderByCol%3Dstatus_sortable%26orderByType%3Dasc)

and others that have the `//legend` on their errors

**Comment by veroglez** [comment]
ci:test:forms

**Comment by veroglez** [comment]
ci:test:data-engine

**Comment by veroglez** [comment]
ci:test:relevant

**Comment by veroglez** [comment]
ci:test:relevant

**Comment by veroglez** [comment]
ci:test:forms

**Comment by rebsilva** [comment]
ci:test:sf

**Comment by veroglez** [comment]
Hi @rebsilva! analyzing the failures, they seem to be not related. Can we move forward the PR? Thanks in advance!

**Comment by rebsilva** [comment]
i will take a look at the tests later today!

**Comment by rebsilva** [comment]
hi @marcelabc! there are some BE tests failing on this PR that seem related to @veroglez changes, could you please help her on what she should do next?

i'm on vacations starting now so if new changes on the FE are made, another person will be needed to review this PR since i won't be available until march 7th. thanks in advance for you help! :purple_heart: also thanks @veroglez for your patience, and i hope this PR will be forwarded soon :cherry_blossom:

**Comment by veroglez** [comment]
Hi @rebsilva, I have added a new commit https://github.com/liferay-forms/liferay-portal/pull/2459/commits/c8951f296b2e9be1f1cdeb8a4811deb48314abf7. I think this commit can solve the failing tests you mention, thanks for your help!

**Comment by veroglez** [comment]
ci:test:data-engine

**Comment by veroglez** [comment]
ci:report:2539928771

**Comment by jkappler** [comment]
Hi @marcelabc, since @rebsilva is out on vacation, could anyone else please help us out with this?

Thanks in advance ❤️

**Comment by marcelabc** [comment]
I reviewed the CI results and it is LGTM :tada:

**Comment by marcelabc** [comment]
ci:forward:force

**Comment by veroglez** [comment]
ci:forward:force

**Comment by jkappler** [comment]
thanks a lot @marcelabc!

**Comment by jkappler** [comment]
ci:test:stable

**Comment by jkappler** [comment]
ci:forward:force

**Pull request closed without merging**